care/facility/api/viewsets/camera_preset.py

```diff
@@ -0,0 +1,48 @@
+from django.shortcuts import get_object_or_404
+from django_filters import rest_framework as filters
+from rest_framework.mixins import ListModelMixin
+from rest_framework.permissions import IsAuthenticated
+from rest_framework.viewsets import GenericViewSet, ModelViewSet
+
+from care.facility.api.serializers.camera_preset import CameraPresetSerializer
+from care.facility.models import AssetBed, CameraPreset
+
+
+class CameraPresetFilter(filters.FilterSet):
+    asset = filters.UUIDFilter(field_name="asset_bed__asset__external_id")
+    bed = filters.UUIDFilter(field_name="asset_bed__bed__external_id")
+    asset_bed = filters.UUIDFilter(field_name="asset_bed__external_id")
+
+
+class AssetBedCameraPresetViewSet(ModelViewSet):
+    serializer_class = CameraPresetSerializer
+    queryset = CameraPreset.objects.all().select_related(
+        "asset_bed", "created_by", "updated_by"
+    )
+    lookup_field = "external_id"
+    permission_classes = (IsAuthenticated,)
+
+    def get_asset_bed_obj(self):
+        return get_object_or_404(
+            AssetBed.objects.filter(external_id=self.kwargs["assetbed_external_id"])
+        )
+
+    def get_queryset(self):
+        asset_bed = self.get_asset_bed_obj()
+        return super().get_queryset().filter(asset_bed=asset_bed)
+
+    def get_serializer_context(self):
+        context = super().get_serializer_context()
+        context["asset_bed"] = self.get_asset_bed_obj()
+        return context
+
+
+class CameraPresetViewSet(GenericViewSet, ListModelMixin):
```

Comment:
what are the requirements for this view?

Reply:
To list all presets that are either associated to a bed or associated to a camera asset (not specific to a single asset-bed link). Required to avoid N+1 API queries from FE when listing all accessible presets in the consultation feed tab and hub camera monitoring page.

Reply:
Should we make two API endpoints instead? `/api/v1/(bed | asset)/{id}/camera-presets/`

Reply:
yeah that would be better, you can reuse the same viewset and filter based on kwargs